client/lib/create-selector/test/index.js

```diff
@@ -277,6 +301,69 @@ describe( 'index', () => {
 			.true;
 	} );
 
+	describe( 'when no custom cache key generating function is given', () => {
+		test( 'should to create a default key by joining args', () => {
+			const memoizedSelector = createSelector( selector );
+
+			memoizedSelector( {}, 'a', 'b', 'c', 1, 2, 3 );
+
+			expect( memoizedSelector.memoizedSelector.cache.has( 'a,b,c,1,2,3' ) ).to.be.true;
+		} );
+	} );
+
+	describe( 'comparing dependants per key', () => {
+		test( 'should clear cache when dependants do not match, but only per key', () => {
+			const state = {
+				a: {
+					b: {
+						c: 'cc',
+						d: 'dd',
+					},
+				},
+			};
+			const memoizedSelector = createSelector(
+				( state, x, y, z ) => state[ x ][ y ][ z ],
+				state => state.a.b
+			);
+
+			const abcFirstResult = memoizedSelector( state, 'a', 'b', 'c' );
+			const abdFirstResult = memoizedSelector( state, 'a', 'b', 'd' );
+
+			const abcSecondResult = memoizedSelector( state, 'a', 'b', 'c' );
+			const abdSecondResult = memoizedSelector( state, 'a', 'b', 'd' );
+
+			state.a.b = {
+				c: 'ccc',
+				d: 'ddd',
+			};
+
+			// should have cached result for both 'abc', and 'abd'
+			expect( memoizedSelector.memoizedSelector.cache.has( 'a,b,c' ) ).to.be.true;
+			expect( memoizedSelector.memoizedSelector.cache.has( 'a,b,d' ) ).to.be.true;
+
+			const abcThirdResult = memoizedSelector( state, 'a', 'b', 'c' );
+			// should have cleared the cache, but only for 'abc', not 'abd'
+
+			expect( memoizedSelector.memoizedSelector.cache.has( 'a,b,c' ) ).to.be.false;
```

Comment:
This assertion is failing -- expected?

Reply:
Yeah, I've not pushed code to meet the tests yet but I'm getting there